src/coordinate_transform/affine_transform.rs

```diff
@@ -0,0 +1,46 @@
+use geo_types::Coord;
+use tiff::{TiffError, TiffFormatError, TiffResult};
+
+use crate::coordinate_transform::CoordinateTransform;
+
+impl CoordinateTransform {
+    pub fn from_transformation_matrix(transformation_matrix: [f64; 16]) -> TiffResult<Self> {
+        let mut transform = [0f64; 6];
+        transform[0] = transformation_matrix[0];
+        transform[1] = transformation_matrix[1];
+        transform[2] = transformation_matrix[3];
+        transform[3] = transformation_matrix[4];
+        transform[4] = transformation_matrix[5];
+        transform[5] = transformation_matrix[7];
```

Comment:
Slight nit: since it doesn't look like you actually mutate `transform` anywhere after initialization, I find it a bit easier to read if you use
```rs
let transform = [
transformation_matrix[0],
transformation_matrix[1],
transformation_matrix[3],
transformation_matrix[4],
transformation_matrix[5],
transformation_matrix[7]
];
```
because then I don't have to guess whether `transform` will be mutated or not later on.

(Ditto for `inverse_transform`)

Reply:
Side tangent: I'm thinking if we could actually just use https://docs.rs/geo/0.28.0/geo/algorithm/affine_ops/struct.AffineTransform.html instead of re-implementing things here.

Reply:
> Slight nit: since it doesn't look like you actually mutate `transform` anywhere after initialization, I find it a bit easier to read if you use
> 
> ```rust
> let transform = [
> transformation_matrix[0],
> transformation_matrix[1],
> transformation_matrix[3],
> transformation_matrix[4],
> transformation_matrix[5],
> transformation_matrix[7]
> ];
> ```
> 
> because then I don't have to guess whether `transform` will be mutated or not later on.
> 
> (Ditto for `inverse_transform`)

Agree, changed in 23b7ac0ec38e284e5e6112b1222e97b4bb47a70b.

Reply:
> Side tangent: I'm thinking if we could actually just use https://docs.rs/geo/0.28.0/geo/algorithm/affine_ops/struct.AffineTransform.html instead of re-implementing things here.

I thought about it too but decided against it to avoid including ´geo´ as a dependency. I'm not sure whether the costs outweigh the benefits.

Reply:
> > Side tangent: I'm thinking if we could actually just use https://docs.rs/geo/0.28.0/geo/algorithm/affine_ops/struct.AffineTransform.html instead of re-implementing things here.
> 
> I thought about it too but decided against it to avoid including ´geo´ as a dependency. I'm not sure whether the costs outweigh the benefits.

Yeah, I feel like the affine transform ops could be split out into its own thing (the gdal crate could probably use some of this too). Ok to keep things as is for now, and maybe keep an eye on https://github.com/georust/geo/issues/1186 if `geo-types` ever merges with `geo`.

Reply:
Ok, will mark this one as resolved, then.